### PR TITLE
Prevent automatic keyword linking in pullquotes

### PR DESCRIPTION
**Pull request by TBonnin**
cc @jennysivapalan @rich-nguyen

### CODE REVIEW

File: common/app/views/support/HtmlCleaner.scala [hunk 1]
```diff
@@ -429,7 +429,11 @@ case class TagLinker(article: Article)(implicit val edition: Edition, implicit v
         val unlinkedParas = paragraphs.filterNot(_.html.contains("<a"))
 
         // pre-filter paragraphs so we do not do multiple regexes on every single paragraph in every single article
-        val candidateParagraphs = unlinkedParas.filter(_.html.contains(keyword.name))
+        val containsKeywordParas = unlinkedParas.filter(_.html.contains(keyword.name))
+
+        // don't link tags contained in article's pullquotes
+        val isPullquote = (e: Element) => (e.tagName() == "aside" && e.hasClass("element-pullquote"))
+        val candidateParagraphs = containsKeywordParas.filterNot(_.parents.exists(ancestor => isPullquote(ancestor)))
```
Comment (TBonnin) [inline common/app/views/support/HtmlCleaner.scala]: I am not sure about the performance implications here. What do you think @jennysivapalan?

Comment (rich-nguyen) [inline common/app/views/support/HtmlCleaner.scala]: Maybe you could filter the paragraphs at line 422?

**Comment by TBonnin** [comment]
Before:
before


After:
after

**Comment by jennysivapalan** [comment]
:+1: :speech_balloon: Great - let me know when you are ready to get this out and I can go through it with you if you like.

**Pull request merged**